settings modal styling

shriyan44 opened the pull request.

shriyan44 committed:
fixiing styling

shriyan44 committed:
finished settings modal

new colors

shriyan44 committed:
Update components/Profile/SettingsModal.jsx

shriyan44 committed:
Merge branch 'main' of https://github.com/jpoist97/GainChanger into settings-modal

fixed conflicts

jpoist97 commented on components/Profile/SettingsModal.jsx:
```diff
@@ -107,10 +110,43 @@ const SettingsModal = (props) => {
   });
   const [selectedColor, setSelectedColor] = useState(settings.colorTheme);
```
Is there a reason we need this if we have it in the settingsState already?

jpoist97 commented on components/Profile/SettingsModal.jsx:
```diff
@@ -107,10 +110,43 @@ const SettingsModal = (props) => {
     // If changes were made
-    if (settings.enableRestNotifications !== settingState.enableRestNotifications || `${settings.restNotificationTimer}` !== settingState.restNotificationTimer || settings.colorTheme !== settingState.colorTheme) {
-      Alert.alert('You have unsaved changes, please discard or save them!');
+    if (settings.enableRestNotifications !== settingState.enableRestNotifications || `${settings.restNotificationTimer}` !== settingState.restNotificationTimer) {
```
Can you add a check for colorTheme here, right now if you change color theme and then close the modal the alert doesn't show

jpoist97 commented on components/Profile/SettingsModal.jsx:
```diff
@@ -107,10 +110,43 @@ const SettingsModal = (props) => {
+    });
+
+    setModalVisible(!modalVisible);
+    console.log('SAVED');
```
```suggestion
```

jpoist97 commented on components/Profile/SettingsModal.jsx:
```diff
@@ -107,10 +110,43 @@ const SettingsModal = (props) => {
+      restNotificationTimer: `${settings.restNotificationTimer}`,
+    });
+    setModalVisible(!modalVisible);
+    console.log('discard');
```
```suggestion
```

jpoist97 commented on components/Profile/SettingsModal.jsx:
```diff
@@ -107,10 +110,43 @@ const SettingsModal = (props) => {
+    if (settings.enableRestNotifications !== settingState.enableRestNotifications || `${settings.restNotificationTimer}` !== settingState.restNotificationTimer) {
+      Alert.alert('Unsaved Changes', 'You have unsaved changes, do you want to keep or discard them?', [{
+        text: 'Keep',
+        onPress: () => { saveChanges(); },
```
nit: 
```suggestion
        onPress: saveChanges,
```

jpoist97 commented on components/Profile/SettingsModal.jsx:
```diff
@@ -184,13 +221,45 @@ const SettingsModal = (props) => {
+                  source={require('../../assets/purple.png')}
+                  style={{ width: 130, height: 30 }}
+                />
+              </ColorButton>
```
Can we use percentages on these button widths so this component looks good on different devices

jpoist97 reviewed (commented):
Made a couple comments, should be good to go in soon. Also I'm not sure how I feel about how we currently show users that they've selected a color theme, but I'll let the group decide

shriyan44 committed:
changing colorTheme save state

shriyan44 committed:
fins changes for home logo

shriyan44 committed:
fixed modal styling

- made "set a cycle" card grey
- added logos to home page
- made color change work

shriyan44 committed:
fixed lint

jpoist97 reviewed (commented):
Code for modal looks good, I'll let Fin review the color stuff since he knows how it works. Only comment I have is it looks like there is more padding on the right side of the log out button than on the left side. Also I think it'd look better if the buttons were centered, but up to the team

finlaylp committed:
removed a bit of duplicate code

finlaylp commented on components/Home/Home.jsx:
```diff
@@ -80,11 +80,11 @@ export default () => {
         style={{
-          width: 215,
-          height: 215,
+          width: 230,
```
You could use Dimensions.Window here so that it changes with diff screen sizes

finlaylp commented on components/Profile/SettingsModal.jsx:
```diff
@@ -179,18 +204,86 @@ const SettingsModal = (props) => {
+                    colorTheme: 'aqua',
+                  });
+                }}
+                style={{ width: '45%', height: 32 }}
```
If we need to get rid of more duplicate code could put this style into the Styled component thingy

finlaylp reviewed (dismissed):
Looks good. We can change the dimensions later

finlaylp committed:
took out like 4 lines of duplicate

finlaylp committed:
forgot to lint
